File: src/main/java/com/example/eyeofsauron/service/MarkerReaderService.java
package com.example.eyeofsauron.service;

import com.example.eyeofsauron.entity.MarkerReader;
import com.example.eyeofsauron.repository.MarkerReaderRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class MarkerReaderService {
    private MarkerReaderRepository markerReaderRepository;

    public List<MarkerReader> getAll() { return markerReaderRepository.findAll(); }

    public Optional<MarkerReader> getById(Long id) { return markerReaderRepository.findById(id); }

    public void create(MarkerReader markerReader) { markerReaderRepository.save(markerReader); }

    public void update(MarkerReader markerReader) { markerReaderRepository.save(markerReader); }

    public void delete(MarkerReader markerReader) { markerReaderRepository.delete(markerReader); }

    public void deleteById(Long id) { markerReaderRepository.deleteById(id); }
}

File: src/main/kotlin/com/example/eyeofsauron/service/MarkerReaderService.kt
package com.example.eyeofsauron.service

import com.example.eyeofsauron.entity.MarkerReader
import com.example.eyeofsauron.repository.MarkerReaderRepository
import org.springframework.stereotype.Service

/**
 * Сервис для работы с устройствами для считывания меток
 * @author rodkinsi
 */
@Service
class MarkerReaderService(private val repository: MarkerReaderRepository) {
    fun getAll(): List<MarkerReader> = repository.findAll()

    fun getById(id: Long) = repository.findById(id)

    fun getByImei(imei: String) = repository.findByImei(imei)

    fun create(markerReader: MarkerReader) = repository.save(markerReader)

    fun update(markerReader: MarkerReader) = repository.save(markerReader)

    fun delete(markerReader: MarkerReader) = repository.delete(markerReader)

    fun deleteById(id: Long) = repository.deleteById(id)
}